crates/router/src/connector/klarna/transformers.rs

```diff
@@ -157,12 +160,18 @@ impl TryFrom<types::PaymentsResponseRouterData<KlarnaPaymentsResponse>>
     ) -> Result<Self, Self::Error> {
         Ok(Self {
             response: Ok(types::PaymentsResponseData::TransactionResponse {
-                resource_id: types::ResponseId::ConnectorTransactionId(item.response.order_id),
+                resource_id: types::ResponseId::ConnectorTransactionId(
+                    item.response.order_id.clone(),
+                ),
                 redirection_data: None,
                 mandate_reference: None,
                 connector_metadata: None,
                 network_txn_id: None,
-                connector_response_reference_id: None,
+                connector_response_reference_id: Some(
+                    item.response
+                        .merchant_reference1
+                        .unwrap_or(item.response.order_id),
+                ),
```

Comment:
```suggestion
                connector_response_reference_id: item.response.merchant_reference1.or(Some(item.response.order_id)),
```
This way we can reduce one operation when merchant_reference1 is not none .